bindings/python/opendaq/generated/reader/py_stream_reader.cpp

```diff
@@ -25,73 +14,255 @@
  * limitations under the License.
  */
 
-#include "py_opendaq/py_opendaq.h"
 #include "py_core_types/py_converter.h"
+#include "py_opendaq/py_opendaq.h"
 
-#include <pybind11/stl.h>
 #include <pybind11/chrono.h>
+#include <pybind11/stl.h>
+
+using SampleTypeVariant = std::variant<py::array_t<daq::SampleTypeToType<daq::SampleType::Float32>::Type>,
+                                       py::array_t<daq::SampleTypeToType<daq::SampleType::Float64>::Type>,
+                                       py::array_t<daq::SampleTypeToType<daq::SampleType::UInt32>::Type>,
+                                       py::array_t<daq::SampleTypeToType<daq::SampleType::Int32>::Type>,
+                                       py::array_t<daq::SampleTypeToType<daq::SampleType::UInt64>::Type>,
+                                       py::array_t<daq::SampleTypeToType<daq::SampleType::Int64>::Type>,
+                                       py::array_t<daq::SampleTypeToType<daq::SampleType::UInt8>::Type>,
+                                       py::array_t<daq::SampleTypeToType<daq::SampleType::Int8>::Type>,
+                                       py::array_t<daq::SampleTypeToType<daq::SampleType::UInt16>::Type>,
+                                       py::array_t<daq::SampleTypeToType<daq::SampleType::Int16>::Type>>;
+
+using DomainTypeVariant = SampleTypeVariant;
+
+template <typename ValueType>
+inline py::array_t<ValueType> read(const daq::StreamReaderPtr& reader, size_t count, size_t timeoutMs)
+{
+    std::vector<ValueType> values(count);
+    reader->read(values.data(), &count, timeoutMs);
+    values.resize(count);
+    return toPyArray(std::move(values));
+}
+
+template <typename ValueType, typename DomainType, typename ReaderType>
+inline std::tuple<SampleTypeVariant, DomainTypeVariant> read(const ReaderType& reader, size_t count, size_t timeoutMs)
+{
+    using DomainVectorType = typename std::conditional<std::is_same<DomainType, std::chrono::system_clock::time_point>::value,
+                                                       std::vector<int64_t>,
+                                                       std::vector<DomainType>>::type;
+
+    std::vector<ValueType> values(count);
+    DomainVectorType domain(count);
+
+    if constexpr (std::is_same_v<DomainType, std::chrono::system_clock::time_point>)
+    {
+        static_assert(sizeof(std::chrono::system_clock::time_point::rep) == sizeof(int64_t));
+        reader.readWithDomain(values.data(), reinterpret_cast<std::chrono::system_clock::time_point*>(domain.data()), &count, timeoutMs);
+    }
+    else
+    {
+        reader.readWithDomain(values.data(), domain.data(), &count, timeoutMs);
+    }
+
+    domain.resize(count);
```

Comment:
Why is resize to the same size (count) needed?

Reply:
> Why is resize to the same size (count) needed?

Final vectors are resized to read size

Reply:
> Why is resize to the same size (count) needed?

From what I understand, `count` is both input and output parameter, right? So if you request reading 100 samples when only 10 are available, the function will reserve an array for 100 samples and later resize it to 10 once it knows the actual number that have been read.

Reply:
Yes, of course.